pyHS100/antitheft.py

```diff
@@ -0,0 +1,199 @@
+"""
+pyHS100
+Python library supporting TP-Link Smart Plugs/Switches (HS100/HS110/Hs200).
+
+The communication protocol was reverse engineered by Lubomir Stroetmann and
+Tobias Esser in 'Reverse Engineering the TP-Link HS110':
+https://www.softscheck.com/en/reverse-engineering-tp-link-hs110/
+
+This library reuses codes and concepts of the TP-Link WiFi SmartPlug Client
+at https://github.com/softScheck/tplink-smartplug, developed by Lubomir
+Stroetmann which is licensed under the Apache License, Version 2.0.
+
+You may obtain a copy of the license at
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+
+class AntiTheft(object):
+    def __init__(self, _device, _query_helper):
+        self._device = _device
+        self._query_helper = _query_helper
+
+    @property
+    def _get_antitheft(self):
+        """
+        Gets all rules.
+
+        :return: list rules
+        :rtype: list
+        :raises SmartPlugException: on error
+        """
+
+        return self._query_helper(
+            "anti_theft",
+            "get_rules"
+        )
+
+    def __iter__(self):
+        """
+        Iterates all rules.
+
+        :return: iterator (name, rule)
+        :rtype: iterator
+        :raises SmartPlugException: on error
+        """
+
+        for antitheft in self._get_antitheft:
+            yield antitheft['name'], antitheft
```

Comment:
Would it make sense just to yield from _get_antitheft and let the user to extract the name from the resulting dict? It would make the code more approachable.